### PR TITLE
calibrationd: add function typing

### PR DESCRIPTION
**Pull request by herman5**
**Description**
- Adds mypy coverage to function signatures in calibrationd

**Verification**
- `$ mypy selfdrive/locationd/calibrationd.py` 
- Pre-commit hooks, CI

### CODE REVIEW

File: selfdrive/locationd/calibrationd.py [hunk 1]
```diff
@@ -46,20 +47,20 @@ class Calibration:
   INVALID = 2
 
 
-def is_calibration_valid(rpy):
+def is_calibration_valid(rpy: np.ndarray) -> bool:
```
Comment (adeebshihadeh) [inline selfdrive/locationd/calibrationd.py:50]: can we add the ndarray shape to the type annotation?

Comment (adeebshihadeh) [inline selfdrive/locationd/calibrationd.py:50]: can we add the ndarray shape to the type annotation?

Comment (herman5) [reply selfdrive/locationd/calibrationd.py:50]: Typing support for ndarray shapes is still a work in progress: https://github.com/numpy/numpy/issues/16544

Once that and PEP 646 support for mypy lands (https://github.com/python/mypy/issues/12280), shape annotations could be added if Python is upgraded to 3.11

Comment (adeebshihadeh) [reply selfdrive/locationd/calibrationd.py:50]: ah, cool!

**Pull request merged**